x-pack/packages/kbn-langchain/server/language_models/chat_bedrock_converse/bedrock_runtime_client.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  BedrockRuntimeClient as _BedrockRuntimeClient,
+  BedrockRuntimeClientConfig,
+} from '@aws-sdk/client-bedrock-runtime';
+import { PublicMethodsOf } from '@kbn/utility-types';
+import type { ActionsClient } from '@kbn/actions-plugin/server';
+import { Logger } from '@kbn/logging';
+import { Readable } from 'stream';
+
+export interface CustomChatModelInput extends BedrockRuntimeClientConfig {
+  actionsClient: PublicMethodsOf<ActionsClient>;
+  connectorId: string;
+  logger: Logger;
+  model: string;
+  streaming?: boolean;
+  temperature?: number;
+  maxTokens?: number;
+  anthropicVersion?: string;
+}
+export class BedrockRuntimeClient extends _BedrockRuntimeClient {
+  actionsClient: PublicMethodsOf<ActionsClient>;
+  connectorId: string;
+  logger: Logger;
+  streaming: boolean;
+  temperature?: number;
+  maxTokens?: number;
+  anthropicVersion?: string;
+  constructor({ actionsClient, connectorId, logger, ...fields }: CustomChatModelInput) {
+    super(fields ?? {});
+    this.streaming = fields.streaming ?? true;
+    this.actionsClient = actionsClient;
+    this.connectorId = connectorId;
+    this.logger = logger;
+    this.temperature = fields.temperature;
+    this.maxTokens = fields.maxTokens;
+    this.anthropicVersion = fields.anthropicVersion;
+  }
+  public async send({ input }, options) {
+    const messages = prepareMessages(input.messages);
+    console.log(
+      'subactionparams',
+      JSON.stringify(
+        {
+          ...input,
+          messages,
+        },
+        null,
+        2
+      )
+    );
+    if (this.streaming) {
+      const data = (await this.actionsClient.execute({
+        actionId: this.connectorId,
+        params: {
+          subAction: 'converseStream',
+          subActionParams: {
+            ...input,
+            messages,
+          },
+        },
+      })) as { data: Readable; status: string; message?: string; serviceMessage?: string };
+
+      console.log('stream here', data);
+      if (data.status === 'error') {
+        throw new Error(
+          `ActionsClientBedrockChat: action result status is error: ${data?.message} - ${data?.serviceMessage}`
+        );
+      }
+
+      return {
+        stream: Readable.toWeb(data.data),
```

Comment:
@jacoblee93 this is where I return the stream, is there something else I need to do here?